### PR TITLE
[FEAT] Custom Timestamp function

### PR DESCRIPTION
**Pull request by henilmalaviya**
This PR proposes changes talked in #2.

I have made changes in the types of `withTimestamp`. It can now be a function that returns a string.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -52,11 +53,16 @@ export const logger = (options: LoggerOptions = {}) => {
...
-    return now.toLocaleString();
+  const defaultGetTimestamp = () => {
+    return new Date().toLocaleString();
   };
 
+  // use the `withTimestamp` option if it's a function
+  const getTimestamp =
+    typeof options.withTimestamp === "function"
+      ? options.withTimestamp
+      : defaultGetTimestamp;
```
Comment (tanishqmanuja) [inline src/index.ts]: Can you inline this function to `() => new Date().toLocaleString()` and remove `defaultGetTimestamp`. Would look much cleaner

File: src/index.ts [hunk 2]
```diff
@@ -52,11 +53,16 @@ export const logger = (options: LoggerOptions = {}) => {
...
   if (!enabled) return app;
 
-  const getTimestamp = () => {
-    const now = new Date();
-    return now.toLocaleString();
+  const defaultGetTimestamp = () => {
+    return new Date().toLocaleString();
   };
 
+  // use the `withTimestamp` option if it's a function
```
Comment (tanishqmanuja) [inline src/index.ts]: Remove this comment, code is self explanatory

File: bun.lockb
Comment (tanishqmanuja) [inline bun.lockb:1]: Please remove this file from the PR, if possible as it is not a good practice to commit lockfile changes in a PR that doesn`t update package.json

**Review by tanishqmanuja — CHANGES_REQUESTED** [review]

**Comment by tanishqmanuja** [comment]
Thanks for the PR, looks great. Also the code style matches with `withBanner` which is a huge positive 💯.

File: bun.lockb
Comment (henilmalaviya) [reply bun.lockb:1]: yea! my bad, my first contribution to repo involving a new feature.

Comment (henilmalaviya) [reply bun.lockb:1]: oh wait did i had to delete the file! fuk i had to restore the lock file no?

**Comment by tanishqmanuja** [comment]
Accidentally removed lockfile from the worktree instead of PR 😅

**Comment by henilmalaviya** [comment]
oh finally! LOL. :sweat_smile:

**Pull request merged**